src/protocol/attribution/credit_capping.rs

```diff
@@ -284,50 +288,67 @@ where
     //     current_credit
     //   }
 
-    try_join_all(
-        (0..num_rows)
-            .zip(repeat((ctx, cap)))
-            .map(|(i, (ctx, cap))| async move {
-                if i == num_rows - 1 {
-                    return Ok(original_credits[i].clone());
-                }
-                let record_id = RecordId::from(i);
-
-                let original_credit = &original_credits[i];
-                let next_prefix_summed_credit = &prefix_summed_credits[i + 1];
-                let current_prefix_summed_credit_exceeds_cap = &exceeds_cap_bits[i];
-                let next_credit_exceeds_cap = &exceeds_cap_bits[i + 1];
-                let next_event_has_same_match_key = &input[i + 1].helper_bit;
-
-                let remaining_budget = if_else(
-                    ctx.narrow(&Step::IfNextEventHasSameMatchKeyOrElse),
+    let capped = zip(
```

Comment:
This is the most interesting item.  We can't do what we did before, which had the processing for the last record return immediately.  That violates the seq_join contract, and I don't want to append to the vector and maybe reallocate.  So I've used seq_join here directly.